lib/ansible/utils/collection_loader/_collection_finder.py

```diff
@@ -531,6 +531,13 @@ def _validate_final(self):
             # only search within the first collection we found
             self._subpackage_search_paths = [self._subpackage_search_paths[0]]
 
+    def exec_module(self, module):
+        collection_name = '.'.join(self._split_name[1:3])
+        AnsibleCollectionConfig.on_collection_load.fire(collection_name=collection_name, collection_path=os.path.dirname(module.__file__))
```

Comment:
I think, for this, we basically just need the `exec` stuff from `load_module`.  But I haven't tested all of the pieces put together at the same time, so I could be wrong.